report_generator/report_generator.py

```diff
@@ -3,24 +3,44 @@
 
 
 class ReportGenerator:
+    """
+    This class is responsible for filtering and generating
+    a report from a collection of imported csv statments.
+    """
+
     def __init__(self) -> None:
         self.transactions: List[Transaction] = []
+        self.total_expense = 0.0
+        self.total_income = 0.0
+        self.ignore_payment = 0.0
 
     def import_transactions(self, transactions: List[Transaction]):
+        """
+        Store and accumulate lists of transaction
+        objects for filtering
+        """
         self.transactions += transactions
 
-    def generate_expense_report(self):
-        total_income = 0
-        total_expense = 0
+    def filter_transations(self):
+        """
+        Scan for key words
+        in attributes of the transaction object
+        and add them to the Report Generator
+        class attributes.
```

Comment:
They're not class attributes, so suggest removing word 'class'

```suggestion
        attributes.
```